paasta_tools/deployd/watchers.py

```diff
@@ -305,8 +307,13 @@ def process_default(self, event):
         self.watch_new_folder(event)
         event = self.filter_event(event)
         if event:
+            if '.json' in event.name and event.path.split('/')[-1] == 'secrets':
```

Comment:
given that you've done this twice, it might be worth extracting a ``def service_for_event`` function to encapsulate this logic.

Reply:
I think this is a good thing to do and I'm going to make it contain the filtering logic too.